stackdio/api/stacks/utils.py

```diff
@@ -97,6 +118,71 @@ def delete_map(self, query='list_nodes'):
         """
         return super(StackdioSaltCloudMap, self).delete_map(query)
 
+    def get_running_by_names(self, names, query='list_nodes', cached=False, profile=None):
+        """
+        Override this so we only get appropriate things for our map
+        """
+        if isinstance(names, basestring):
+            names = [names]
+
+        matches = {}
+        handled_drivers = {}
+        mapped_providers = self.map_providers_parallel(query, cached=cached)
+        for alias, drivers in mapped_providers.items():
+            for driver, vms in drivers.items():
+                if driver not in handled_drivers:
+                    handled_drivers[driver] = alias
+                # When a profile is specified, only return an instance
+                # that matches the provider specified in the profile.
+                # This solves the issues when many providers return the
+                # same instance. For example there may be one provider for
+                # each availability zone in amazon in the same region, but
+                # the search returns the same instance for each provider
+                # because amazon returns all instances in a region, not
+                # availability zone.
+                if profile:
+                    if alias not in self.opts['profiles'][profile]['provider'].split(':')[0]:
+                        continue
+
+                for vm_name, details in vms.items():
+                    # XXX: The logic below can be removed once the aws driver
+                    # is removed
+                    if vm_name not in names:
+                        continue
+
+                    elif driver == 'ec2' and 'aws' in handled_drivers and \
```

Comment:
This should probably use parens instead of backslashes for multi-line conditions